wasm-transform/src/machine.rs

```diff
@@ -395,13 +395,16 @@ impl<I: TryFromImport, R: RunnableCode> Artifact<I, R> {
         &self,
         host: &mut H,
         name: &Q,
+        fallback_name: Option<&Q>,
```

Comment:
I am strongly against doing this. The machine does not know, nor should it know, about "fallback names".

It is responsible for running an entrypoint. It is the caller that should decide which entrypoint to run. This is how it is when the scheduler invokes it, and this is how it should be when cargo-concordium does.

The reason for this is that there is clear separation and the semantics of Wasm are not intertwined with the semantics of V0/V1 contracts.

So please do not handle the fallback logic in the interpreter, but rather in `wasm-chain-integration/v1/mod.rs` in the invoke_receive method, or in `cargo-concordium` itself.

Reply:
Doing this in cargo-concordium also gives you the opportunity to tell the user that this is being done.